docker/client/container.sls

```diff
@@ -6,12 +6,34 @@ include:
 {%- for name, container in client.get('container', {}).iteritems() %}
   {%- set id = name %}
   {%- set required_containers = [] %}
+{#
+get_docker_py:
+  cmd.run:
+  - name: 'apt-get -y install python-pip --force-yes && pip install docker-py'
+  - reload_modules: true
+#}
+{%- if not grains.get('oscodename', 'trusty') %}
+
+docker_compose_python_pip:
+  pkg.installed:
+    - name: python-pip
+    - reload_modules: true
 
 {{id}}_image:
   dockerng.image_present:
     - name: {{ container.image }}
+    - reload_modules: true
     - require:
-      - pkg: docker_python
+      - pkg: python-docker
+
+{%- else %}
+
+{{id}}_image:
+  dockerng.image_present:
+    - name: {{ container.image }}
+    - reload_modules: true
```

Comment:
Why are you adding reload modules everywhere? That will just slow down execution imho.